file_transfer_agent_test.go

```diff
@@ -597,3 +599,31 @@ func TestUploadWhenFilesystemReadOnlyError(t *testing.T) {
 		t.Fatalf("should error when creating the temporary directory. Instead errored with: %v", err)
 	}
 }
+
+func TestUnitUpdateProgess(t *testing.T) {
```

Comment:
Good one! Can I suggest a small refactoring here? Maybe we can move `percent` and `updateProgress` functions to be functions on `spp` type? They are strongly connected.